internal/stores/gcs.go

```diff
@@ -119,6 +115,9 @@ func (s *GCSStore) Upload(ctx context.Context, objects ...string) error {
 			if strings.Contains(err.Error(), "conditionNotMet") {
 				logger.Infof("%s already exists, skipping...", o)
 			} else {
+				if err := cleanup(); err != nil {
```

Comment:
We are getting 3 deep on if statements, Is there some other way we could structure this? See comment on gcs.go line 99 for a possible idea. I'm not sure on the best pattern.

Reply:
Let me know if multierror makes it better or worse

Reply:
Better! I like it. Keeps the other error from being buried.